avm/res/api-management/service/backend/main.bicep

```
@@ -67,3 +80,72 @@ output name string = backend.name
 
 @sys.description('The resource group the API management service backend was deployed into.')
 output resourceGroupName string = resourceGroup().name
+
+// ================ //
+// Definitions      //
+// ================ //
+
+type backendCredentialsType = {
+  @sys.description('Optional. Authorization header authentication.')
+  authorization: resourceInput<'Microsoft.ApiManagement/service/backends@2024-05-01'>.properties.credentials.authorization?
+  @sys.description('Optional. List of Client Certificate Thumbprints. Will be ignored if certificatesIds are provided.')
+  certificate: string[]?
+  @sys.description('Optional. List of Client Certificate Ids.')
+  certificateIds: string[]?
+  // Each property within header is an array of strings in format
+  // {customized property}: [
+  //   'string'
+  // ]
+  @sys.description('Optional. Header Parameter description.')
+  header: object?
+  // Each property within query is an array of strings in format
+  // {customized property}: [
+  //   'string'
+  // ]
+  @sys.description('Optional. Query Parameter description.')
+  query: object?
+}
+
+type backendProxyType = {
+  @secure()
+  @sys.description('Optional. Password to connect to the WebProxy Server.')
+  password: string?
+  @sys.description('Required. WebProxy Server AbsoluteUri property which includes the entire URI stored in the Uri instance, including all fragments and query strings.')
```

For readability, I'd recommend to leave some space in between individual parameter/property-declarations.

```suggestion

  @sys.description('Required. WebProxy Server AbsoluteUri property which includes the entire URI stored in the Uri instance, including all fragments and query strings.')
```